lib/Dancer2/Core/App.pm

```diff
@@ -1619,6 +1620,33 @@ sub response_not_found {
     return $response;
 }
 
+sub uri_for_route {
+    my ( $self, $route_name, $named_params, $unnamed_params, $dont_escape ) = @_;
+
+    my $route =
+      List::Util::first { $_->name eq $route_name } @{$self->routes->{'get'}};
```

Comment:
Taking inspiration from `any`, would it be worth accepting an arrayref of `[$method, $name]` for `$route_name`?  Then we could support POST with no extra effort.

Reply:
I think this is reasonable. Let me think of how I want to support other methods before I commit to the semantics of `any`.

Reply:
If we add a check for the route by name, it won't matter if the route is a GET, PUT, POST, or anything else.

We also don't produce anything other than a URI, so whether it will be used in a GET or POST method is up to the user.

Reply:
Resolved